client/client.cpp

```diff
@@ -424,120 +426,131 @@ void* read_poll_can(void* args)
 
     while (poll)
     {
-        if (count > (BUF_SZ - frame_sz))
-        {
-            //full buffer, drop data and start over. TODO: ring buffer, print/debug
-            bufpnt = read_buf_can;
-            count = 0;
-        }
-
 #if CAN_FORWARDER_MODE
-        int num_bytes_can = read_frame(fd, &frame);
+        int num_bytes_can = read_frame(fd, &frame, count != 0);
 #else
-        int num_bytes_can = read_frame(fd, &frame, &tv);
+        int num_bytes_can = read_frame(fd, &frame, count != 0, &tv);
 #endif
         if (num_bytes_can == -1)
         {
             // This happens when there is a timeout, we simply keep looping, as we want do not want to block while
             // reading the CAN-Bus, as then we would never be able to shut down the threads if there was no activity
             // on the bus
-            continue;
+        }
+        else if (num_bytes_can < -1)
```

Comment:
Do we really need this check. `recv` should not return anything below -1 according to the spec: https://man7.org/linux/man-pages/man2/recv.2.html

Reply:
It is defensive programming of validating inputs from external code. The code in the else case does not handle negative values. I was treating the pthread_error as an assert that is active in release builds.

I could merge it with the previous if to have `if(num_bytes_can < 0)` since that branch is okay with negative numbers.